openwhisk/README.md

```diff
@@ -0,0 +1,260 @@
+
+These instructions take you through the installation of
+- [x] [Istio v1.0.7](https://github.com/istio/istio/releases) using resources specifically configured for use with Knative Serving
+- [x] [Knative v0.5.0](https://github.com/knative/serving/releases/tag/v0.5.0)
+
+## Verify Knative installation
+
+#### Verify Istio pods are running
+
+```bash
+$ kubectl get pods --namespace istio-system
+```
+<details>
+    <summary>Sample output</summary>
+
+```
+NAME                                       READY     STATUS      RESTARTS   AGE
+cluster-local-gateway-547467ccf6-p8n72     1/1       Running     1          4d21h
+istio-citadel-7d64db8bcf-m7gsj             1/1       Running     0          4d21h
+istio-cleanup-secrets-8lzj4                0/1       Completed   0          4d21h
+istio-egressgateway-6ddf4c8bd6-2dxhc       1/1       Running     1          4d21h
+istio-galley-7dd996474-pdd6h               1/1       Running     1          4d21h
+istio-ingressgateway-84b89d647f-cxrwx      1/1       Running     1          4d21h
+istio-pilot-86bb4fcbbd-5ns5q               2/2       Running     0          4d21h
+istio-pilot-86bb4fcbbd-vd2xr               2/2       Running     0          4d21h
+istio-pilot-86bb4fcbbd-zstrw               2/2       Running     0          4d21h
+istio-policy-5c4d9ff96b-559db              2/2       Running     1          4d21h
+istio-security-post-install-dv8s7          0/1       Completed   0          4d21h
+istio-sidecar-injector-6977b5cf5b-94hj5    1/1       Running     0          4d21h
+istio-telemetry-7676df547f-jp952           2/2       Running     1          4d21h
+zipkin-8594bbfc6b-z6vdg                    1/1       Running     1          4d21h  
+```
+</details>
+
+#### Verify your default namespace uses Istio for all services
+
+Check the `default` namespace has the label **istio-injection** and it is set to **enabled**:
+
+```bash
+$ kubectl get namespace default -o yaml
+```
+
+Example output:
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  creationTimestamp: "2019-06-13T16:52:46Z"
+  labels:
+    istio-injection: enabled
+  name: default
+  resourceVersion: "18019"
+  selfLink: /api/v1/namespaces/default
+  uid: ab987afe-8dfb-11e9-8092-025000000001
+spec:
+  finalizers:
+  - kubernetes
+status:
+  phase: Active
+```
+
+**Note**: If you do not see the **istio-injection** label, verify you issued the 'kubectl' command to set this label to the default namespace. See [Troubleshooting](#troubleshooting) section for more information.
+
+# Building and Serving OpenWhisk Runtime Tekton Pipeline 
+
+All OpenWhisk Runtime Tekton Pipeline require a valid Kubernetes **Service Account** with access to a Kubernetes **Secret** that contains your Docker Hub username and password.  This credential will be used as part of the Tekton build process to "push" your Knative application image containing your OpenWhisk Action to Docker Hub.
```

Comment:
yup, can work with any image registry as long as Tekton can support it, deleted reference to Docker Hub from here to emphasis the fact that it can push the image to any registry.